test/python/circuit/test_library.py

```diff
@@ -230,6 +230,52 @@ def test_non_symmetric_raises(self, adjacency_matrix):
             GraphState(adjacency_matrix)
 
 
+@ddt
+class TestFourierCheckingLibrary(QiskitTestCase):
+    """Test the Fourier Checking circuit."""
+
+    def assertFourierCheckingIsCorrect(self, f_truth_table, g_truth_table, fc_circuit):
+        """Assert that the Fourier Checking circuit produces the correct matrix."""
+
+        simulated = Operator(fc_circuit)
+
+        num_qubits = int(np.log2(len(f_truth_table)))
+
+        # create Hadamard matrix, re-use approach from TestMCMT
+        h_i = 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
+        h_tot = np.array([1])
+        for _ in range(num_qubits):
+            h_tot = np.kron(h_tot, h_i)
+
+        f_mat = np.diag(f_truth_table)
+        g_mat = np.diag(g_truth_table)
+
+        expected = np.linalg.multi_dot([h_tot, g_mat, h_tot, f_mat, h_tot])
+        expected = Operator(expected)
+
+        self.assertTrue(expected.equiv(simulated))
```

Comment:
Allow me to suggest the following
```suggestion
        self.assertEqual(expected, simulated)
```

Reply:
Thank you for this suggestion, this definitely looks nicer! I have committed this change.

Reply:
This is not the same though, `equiv` neglects a phase difference and `assertEqual` will do a check using `equal` which doesn't neglect phase afaik. Maybe its better to keep `equiv` since a global phase would not matter.

Reply:
Hi @Cryoris, that's a very good point, I haven't thought about it. Thanks for giving this suggestion! I have looked into this just to be sure and I can see that `assertEqual` would call `==` operator, which for `Operator` is `np.allclose`:

https://github.com/Qiskit/qiskit-terra/blob/e265409a4551551feddb4a74928386fd0d27f6df/qiskit/quantum_info/operators/operator.py#L118-L123

`Operator.equiv`, on the other hand, ignores the phase as you have rightly pointed out:

https://github.com/Qiskit/qiskit-terra/blob/e265409a4551551feddb4a74928386fd0d27f6df/qiskit/quantum_info/operators/operator.py#L440-L441

I wanted to write out this explanation just for my own understanding. Thank you for input! I have committed a change which switches back to `equiv`.